src/zulip.rs

```diff
@@ -495,3 +504,45 @@ async fn move_notification(
         .unwrap()),
     }
 }
+
+async fn promote_regression_issues(gh_client: &GithubClient, mut words: impl Iterator<Item = &str>) -> anyhow::Result<()> {
+    let past = match words.next() {
+        Some(past) => Label { name: format!("regression-from-stable-to-{}", past) },
+        None => anyhow::bail!("past state not present")
+    };
+
+    anyhow::ensure!(words.next() == Some("to"), "invalid syntax");
+
+    let current = match words.next() {
+        Some(current) => Label { name: format!("regression-from-stable-to-{}", current) },
+        None => anyhow::bail!("current state not present")
+    };
+
+    let repo = match words.next() {
+        Some(repo) => {
+            if repo.contains("/") {
+                repo.to_string()
+            } else {
+                format!("rust-lang/{}", repo)
+            }
+        },
+        None => "rust-lang/rust".to_string()
+    };
+    let repo = Repository { full_name: repo };
+    
+    let issues = repo
+        .get_issues(&gh_client, &[&past])
+        .await?;
+    let fut = issues
+        .iter()
+        .map(|issue| {
+            let mut labels = issue.labels().to_owned();
+            if let Some(pos) = labels.iter().position(|l| l == &past) {
+                labels.remove(pos);
+            }
+            labels.push(current.clone());
+            issue.set_labels(&gh_client, labels)
+        });
+    futures::future::join_all(fut).await;
```

Comment:
GitHub recommends that we issue requests one at a time -- otherwise they'll aggressively rate limit us. Let's use a combinator of some sort here that does that and spawn it to run asynchronously in the background.

We might look at the APIs, too, I know in the UI you can set a label on a bunch of issues at once. Maybe there's something in the API that can do that as well.

We should make sure we don't block the return from this on the queries completing (we should queue a "post message zulip API call" when it's all done, though).